api.md

```diff
@@ -919,6 +922,53 @@ The expected output:
 
 ---
 
+## interpolate() [](interpolate)
+
+The `interpolate` function does linear interpolation for missing values. It can only be used in an aggregation query with `time_bucket_gapfill`.
+The `interpolate` function call cannot be nested inside other function calls.
+
+#### Required Arguments [](interpolate-required-arguments)
+
+|Name|Description|
+|---|---|
+| `value` | The value to carry forward (anyelement) |
+
+#### Optional Arguments [](interpolate-optional-arguments)
+
+|Name|Description|
+|---|---|
+| `prev` | The lookup expression for values before the gapfill time range (record) |
+| `next` | The lookup expression for values after the gapfill time range (record) |
+
+#### Sample Usage [](interpolate-examples)
+
+Get the temperature every 5 minutes for each device over the last 2 weeks interpolating for missing readings:
+```sql
+SELECT
+  time_bucket_gapfill('5 minutes', time, now() - interval '2 week', now()) as time,
+  device_id,
+  interpolate(avg(temperature))
+FROM metrics
+  WHERE time > now () - interval '2 week'
+GROUP BY 1,2
+ORDER BY 1 DESC;
+```
+
+Get the temperature every 5 minutes for each device over the 2 weeks interpolating for missing readings with lookup queries for values before and after the gapfill time range:
```

Comment:
I don't think the average reader will understand what the before/after queries actually do.  Think you need some text around this explaining the issue before giving the syntax, i.e., the gapfill query defines a time-range, but what if you don't actually have any values at the beginning of the time range, and need to actually search *outside* of the time-range for the right values to interpolate from. This query defines how you perform that search, etc.

I think many of these queries should also have sample output to better explain...

Reply:
I wonder if we want to include this discussion above in the introduction section as well. Think it reads better to present it there, and then show an example here

Reply:
The meaning of those values is explained in the table with the arguments now.